Improve cost of transactions performance section (#516)
neo4j/driver-documentation: python-manual/modules/ROOT/pages/performance.adoc

[[transactions-cost]]
== Be aware of the cost of transactions

Queries are always run within <<transaction, transactions>> on the server, so that the database always ends up in a consistent state regardless of what happens during a transaction execution (power outages, software crashes, etc).

> Queries submitted via xref:query-simple.adoc[`.execute_query()`] or xref:transactions.adoc#managed-transactions[`.execute_read/write()`] are also automatically retried in case of failure.

Reviewer: ```suggestion
Transactions started via xref:query-simple.adoc[`.execute_query()`] or xref:transactions.adoc#managed-transactions[`.execute_read/write()`] are also automatically retried by the driver in case of failure.
```

A transactions yields an overhead that is small, but that adds up as the number of transactions increases.

As your workload grows, evaluate what level of isolation each query needs and be intentional about your usage of transactions.
There are essentially three approaches to running queries depending on your needs for _safety_ and _performance_.
Remember that there's no free lunch and that it's always a trade-off between the two.

1. **One transaction ⟷ One query (lowest throughput, highest safety)** +
The safest mode of execution with respect to failures, because of the isolation and auto-retry guarantees, but also the slowest due to the overhead of transactions scaling with the number of queries.
+
.Run 1000 queries, each as a separate transaction
[source, python]
----
for i in range(1000):
    driver.execute_query("<QUERY>", database_="<database-name>")
    # or session.execute_read/write() calls, each with one query
----

2. **One transaction ⟷ Multiple queries (higher throughput, high safety)** +

> When multiple queries are grouped into a single transaction, the transaction as a whole is isolated from others, whereas individual queries in the transaction are not isolated and failure of one query results in a rollback of all queries.

Reviewer: ```suggestion
When multiple queries are grouped into a single transaction, the transaction as a whole is isolated from others, whereas individual queries in the transaction are not isolated and failure of one query results in a rollback of all queries if it cannot be retried.
```
